File: mycollab-web/src/main/java/com/esofthead/mycollab/module/project/view/parameters/BugScreenData.java
package com.esofthead.mycollab.module.project.view.parameters;

import com.esofthead.mycollab.module.tracker.domain.BugWithBLOBs;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

public class BugScreenData {
	public static class GotoDashboard extends ScreenData {

		public GotoDashboard() {
			super(null);
		}
	}

	public static class Search extends ScreenData<BugSearchParameter> {

		public Search(BugSearchParameter params) {
			super(params);
		}
	}

	public static class Read extends ScreenData<Integer> {

		public Read(Integer params) {
			super(params);
		}
	}
	
	public static class Add extends ScreenData<BugWithBLOBs> {

		public Add(BugWithBLOBs params) {
			super(params);
		}
	}
	
	public static class Edit extends ScreenData<BugWithBLOBs> {

		public Edit(BugWithBLOBs params) {
			super(params);
		}
	}
}

File: mycollab-web/src/main/scala/com/mycollab/module/project/view/parameters/BugScreenData.scala
package com.mycollab.module.project.view.parameters

import com.mycollab.module.tracker.domain.BugWithBLOBs
import com.mycollab.vaadin.mvp.ScreenData

/**
 * @author MyCollab Ltd.
 * @since 5.0.3
 */
object BugScreenData {

    class Read(params: Integer) extends ScreenData[Integer](params) {}

    class Add(params: BugWithBLOBs) extends ScreenData[BugWithBLOBs](params) {}

    class Edit(params: BugWithBLOBs) extends ScreenData[BugWithBLOBs](params) {}

}
